File: workflow/plugins/python/var/var_delete.py
"""Workflow plugin: delete variable from workflow store."""


def run(runtime, inputs):
    """Delete variable from workflow store."""
    key = inputs.get("key")

    if key is None:
        return {"result": False, "deleted": False, "error": "key is required"}

    if key in runtime.store:
        del runtime.store[key]
        return {"result": True, "deleted": True}

    return {"result": False, "deleted": False}

File: workflow/plugins/go/var/var_delete.go
// Package variable provides variable management workflow plugins.
package variable

import (
	plugin "metabuilder/workflow/plugins/go"
)

// Delete removes a variable from the workflow store.
func Delete(runtime *plugin.Runtime, inputs map[string]interface{}) (map[string]interface{}, error) {
	key, ok := inputs["key"].(string)
	if !ok {
		return map[string]interface{}{"success": false, "error": "key is required"}, nil
	}

	_, existed := runtime.Store[key]
	delete(runtime.Store, key)

	return map[string]interface{}{"success": true, "existed": existed}, nil
}
